compiler/plugins/input/StableHLO/Conversion/StableHLOToLinalgExt.cpp

```diff
@@ -427,22 +428,59 @@ struct FftOpConversion final : OpConversionPattern<mlir::stablehlo::FftOp> {
 struct ReverseOpConversion final
     : OpConversionPattern<mlir::stablehlo::ReverseOp> {
   using OpConversionPattern::OpConversionPattern;
-
   LogicalResult
   matchAndRewrite(mlir::stablehlo::ReverseOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     auto ty = dyn_cast<RankedTensorType>(adaptor.getOperands()[0].getType());
     if (!ty)
       return failure();
 
+    Value input = op.getOperand();
+    auto inputTy = cast<ShapedType>(input.getType());
+    auto resultTy = cast<ShapedType>(op.getType());
+    auto dims = op.getDimensions();
     Location loc = op.getLoc();
-    SmallVector<OpFoldResult> mixedSizes =
-        tensor::getMixedSizes(rewriter, loc, adaptor.getOperands()[0]);
-    Value emptyTensor =
-        rewriter.create<tensor::EmptyOp>(loc, mixedSizes, ty.getElementType());
-    rewriter.replaceOpWithNewOp<IREE::LinalgExt::ReverseOp>(
-        op, typeConverter->convertType(op.getType()), adaptor.getOperands(),
-        emptyTensor, rewriter.getI64TensorAttr(op.getDimensions()));
+
+    SmallVector<Value> dynDims;
+    for (int i = 0; i < inputTy.getRank(); i++) {
+      if (inputTy.isDynamicDim(i)) {
+        dynDims.push_back(rewriter.create<tensor::DimOp>(loc, input, i));
+      }
+    }
+
+    // First fill the output buffer with the init value.
+    auto emptyTensor = rewriter
+                           .create<tensor::EmptyOp>(loc, inputTy.getShape(),
+                                                    inputTy.getElementType(),
+                                                    ArrayRef<Value>({dynDims}))
+                           .getResult();
+    SmallVector<AffineMap, 2> affineMaps = {
+        rewriter.getMultiDimIdentityMap(resultTy.getRank())};
+
+    rewriter.replaceOpWithNewOp<linalg::GenericOp>(
+        op, resultTy, ArrayRef<Value>({}), ValueRange{emptyTensor}, affineMaps,
+        getNParallelLoopsAttrs(resultTy.getRank()),
+        [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange args) {
+          llvm::SmallVector<Value> indices;
+          for (unsigned int i = 0; i < inputTy.getRank(); i++) {
+            Value index =
+                rewriter.create<linalg::IndexOp>(nestedLoc, i).getResult();
+            if (std::find(dims.begin(), dims.end(), i) != dims.end()) {
```

Comment:
`dims` is not used til the snippet. Also, can we make it `llvm::SmallSet`, so it is more efficient here?

Reply:
`dims` comes from operand dimensions and is intrinsically an array (dimension order matters there), so probably the code as is is good enough.

Besides, dims is very small so sequential search overhead is likely smaller than a `SmallSet`?

Reply:
Actually, I took a look at the implementation of `llvm::SmallSet`: 
* If `N` (size) is smaller than 32, then `SmallSet` is simply a wrapper of `llvm::SmallVector`.
* If size has grown bigger than threshold then it converts to `std::set`.

So we are good here.

Reply:
I see, this is new to me. Thanks for looking at the details!